File: 485-max-consecutive-ones/main.go
func findMaxConsecutiveOnes(nums []int) int {

    var max_sum int = 0
    var curr_sum int = 0 

    for _, num := range nums {
        if num == 1 {
            curr_sum++
            max_sum = max(max_sum, curr_sum)
        }else {
            curr_sum = 0
        }
    }
    return max_sum

}

func max(a,b int) int {
    if a < b {
        return b
    }
    return a
}

File: 485-max-consecutive-ones/main.rs
impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        
        let (mut curr_sum, mut max_sum) = (0, 0);

        for &num in nums.iter() {
            if num == 1 {
                curr_sum += 1;
                max_sum = std::cmp::max(max_sum, curr_sum);
            }else {
                curr_sum = 0;
            }
        }

        max_sum
    }
}
